test/integration/controlplane/transformation/kms_transformation_test.go

```diff
@@ -706,6 +732,32 @@ resources:
 			if !bytes.HasPrefix(rawEnvelope, []byte(wantPrefix)) {
 				t.Fatalf("expected secret to be prefixed with %s, but got %s", wantPrefix, rawEnvelope)
 			}
+
+			// make sure that old connections have been closed and check again
+			if tc.sleep != 0 {
+				time.Sleep(tc.sleep)
+			}
+			_, err = test.restClient.CoreV1().Secrets("").List(
+				context.TODO(),
+				metav1.ListOptions{},
+			)
+			if err != nil {
+				t.Fatalf("failed to re-list secrets, err: %v", err)
+			}
+
+			// make sure things still work at a "later" time
+			time.Sleep(2 * time.Minute)
```

Comment:
this is going to add a big penalty 8 test-cases = 16 min, can't we do better?

Reply:
I don't intend to merge this as-is, I am trying to root cause an issue I am seeing.

Reply:
oh, I was surprised , that explains it XD